library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java

```diff
@@ -1296,6 +1296,14 @@ public static int getArgbColorFromCeaColor(int red, int green, int blue, int opa
     }
   }
 
+  protected void clearStuckCaptions()
+  {
+    // Do nothing for CEA-708.
+    // As per spec CEA-708 Caption text sequences shall be terminated by either the start of a new
+    // DTVCC Command, or with an ASCII ETX (End of Text) (0x03) character when no other DTVCC
+    // Commands follow.
+  }
```

Comment:
I'm not a fan of having to add a method to CeaDecoder that only makes sense for Cea608.

I wonder (I haven't tested this) if you could add this functionality to `isNewSubtitleDataAvailable()`: If the timeout has expired then set `cues` to `emptyList` - would that work?

Reply:
I thought Sada (@sneelavara) had found that there was a similar issue and fix in CEA-708?

Also, you should know, in our internal review, there was some debate over if it was best to depend on the fact that `dequeueOutputBuffer()` is called every render cycle to run the timer.  Probably not an issue, but I've quoted that review comment in the code.

Reply:
@icbaker isNewSubtitleDataAvailable() is called when there is something to decode on input buffer. That may not be called regularly and we may skip checking for time expiry.

Reply:
@icbaker using `isNewSubtitleDataAvailable()` only works if there is an input buffer, we need the timeout check called most very render() cycle.  So `dequeueOutputBuffer()` is the best method we have without adding a new method.